Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/DriverReferral.hs

```diff
@@ -0,0 +1,42 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# OPTIONS_GHC -Wno-deprecations #-}
+
+module Storage.CachedQueries.DriverReferral where
+
+import Kernel.Prelude
+import qualified Kernel.Storage.Hedis as Hedis
+import Kernel.Utils.Common
+import qualified Storage.Queries.DriverReferral as Queries
+
+getLastRefferalCode :: (CacheFlow m r, EsqDBFlow m r) => m Integer
+getLastRefferalCode =
+  Hedis.safeGet makeLastRefferalCodeKey >>= \case
+    Just a -> do
+      _ <- Hedis.incr makeLastRefferalCodeKey
+      pure a
+    Nothing -> do
+      lastReferralCode <- Queries.getLastRefferalCode
+      cacheLastRefferalCode lastReferralCode
+      _ <- Hedis.incr makeLastRefferalCodeKey
```

Comment:
Return this value instead of (last + 1), there can be race condition here

Reply:
Done